scripts/prep_release.sh

```diff
@@ -16,12 +16,37 @@ SCRIPT_DIR=$(dirname $0)
 SCRIPT_DIR=$(readlink -f $SCRIPT_DIR)
 ROOT_DIR=$(readlink -f "${SCRIPT_DIR}/..")
 
+# Determine the ECR region we want to use
+set +e
+# First attempt to use the EC2 instance meta-data to determine the region
+ECR_REGION="$(curl -fs http://169.254.169.254/latest/meta-data/placement/region)"
+if [ "$?" -ne 0 ]; then
+    # Not running on EC2, so get region from AWS CLI configuration
+    ECR_REGION="$(aws configure get region)"
+    if [ "$?" -ne 0 || -z "${ECR_REGION}" ]; then
+        echo "ERROR: Could not determine region"
+        exit 1
+    fi
+fi
+set -e
+
 # Ensure that the docker daemon is running in the background
 # These commands are currently being done here instead of in buildspec due to the following issue: https://github.com/awslabs/aws-delivlib/issues/448
 # These commands come from the Codebuild sample for running docker on a custom image: https://docs.aws.amazon.com/codebuild/latest/userguide/sample-docker-custom-image.html#sample-docker-custom-image-files
 docker info || nohup /usr/bin/dockerd --host=unix:///var/run/docker.sock --host=tcp://127.0.0.1:2375 --storage-driver=overlay2 &
 timeout 15 sh -c "until docker info; do echo .; sleep 1; done"
 
+# Pull amazonlinux image locally from ECR. This is done to avoid pulling from DockerHub both for egress and rate
+# limiting.
+PULL_AL_FROM_ECR_ARGS=(
+    # Region of ECR repo
+    "${ECR_REGION}"
+    # Image versions to pull
+    "latest" # required for building and publishing lambda layers
+    "2"      # required for building Deadline docker images for running integration tests
```

Comment:
The integration tests don't use the latest version of the AL2 image? Any particular reason?

Reply:
The Deadline Docker recipes pin to use the major version 2 of the `amazonlinux` image:

```docker
FROM amazonlinux:2 AS builder
```

Technically, the integration tests don't yet run in the release automation, but #205 is looking to change that. This is a proactive step to have that image pulled locally from ECR so that building staged recipes will not be throttled when that is integrated.